classroom/snippets/src/main/java/DeleteInvitation.java

```diff
@@ -0,0 +1,72 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START classroom_delete_invitation]
+
+import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport;
+import com.google.api.client.googleapis.json.GoogleJsonError;
+import com.google.api.client.googleapis.json.GoogleJsonResponseException;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.gson.GsonFactory;
+import com.google.api.services.classroom.Classroom;
+import com.google.api.services.classroom.ClassroomScopes;
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.util.ArrayList;
+import java.util.Arrays;
+
+/* Class to demonstrate the use of Classroom Delete Invitation API. */
+public class DeleteInvitation {
+
+  /* Scopes required by this API call. If modifying these scopes, delete your previously saved
+  tokens/ folder. */
+  static ArrayList<String> SCOPES = new ArrayList<>(
+      Arrays.asList(ClassroomScopes.CLASSROOM_ROSTERS));
+
+  /**
+   * Deletes an invitation.
+   *
+   * @param id - the identifier of the invitation to delete.
+   * @throws IOException - if credentials file not found.
+   * @throws GeneralSecurityException - if a new instance of NetHttpTransport was not created.
+   */
+  public static void deleteInvitation(String id) throws GeneralSecurityException, IOException {
+
+    // Create the classroom API client.
+    final NetHttpTransport HTTP_TRANSPORT = GoogleNetHttpTransport.newTrustedTransport();
+    Classroom service =
+        new Classroom.Builder(
+            HTTP_TRANSPORT,
+            GsonFactory.getDefaultInstance(),
+            ClassroomCredentials.getCredentials(HTTP_TRANSPORT, SCOPES))
+            .setApplicationName("Classroom samples")
+            .build();
+
+    // [START classroom_delete_invitation_code_snippet]
+    try {
+      service.invitations().delete(id).execute();
```

Comment:
it's interesting that the `invitations` service supports `delete` and not `patch`, while `guardianInvitations` don't support `delete` and do support `patch`, and `patch` must be used for `delete`s https://developers.google.com/classroom/reference/rest/v1/userProfiles.guardianInvitations/patch. No action here but this might be worth a note or something when you get to writing the guide, since I find it kind of confusing and inconsistent myself.

Reply:
That is good feedback. I made a note to include clarity in the Devsite guides about this when I get to updating them.